src/server/models/listItem.ts

```diff
@@ -195,6 +204,143 @@ export async function checkListItemExists({
   return total > 0;
 }
 
+export async function getPaginationValues(  props: {
+  count: number;
+  page: number;
+  params: ListsRequestParams;
+}): Promise<PaginationResults> {
+  const { count, page, params } = props;
+  const limit = 20;
+  let allPages = 0;
+  if (count > 0 && limit > 0) {
+    allPages = Math.ceil(count / limit);
+  }
+
+  const nextPrevious = await getNextPrevious({page, allPages, params});
+  const { queryString, currentPage } = nextPrevious;
+
+  const pageItems: PaginationItem[] = await getPageItems({allPages, currentPage, queryString});
+
+  const from = await getFromCount({count, limit, currentPage});
+  const to = await getToCount({currentPage, allPages, count, limit});
+
+  return {
+    pagination: {
+      results: {
+        from,
+        to,
+        count,
+        currentPage
+      },
+      previous: {
+        text: nextPrevious.previous.page.toString(),
+        href: nextPrevious.previous.queryString
+      },
+      next: {
+        text: nextPrevious.next.page.toString(),
+        href: nextPrevious.next.queryString
+      },
+      items: pageItems
+    }
+  }
+}
+
+ async function getNextPrevious(props: {page: number; allPages: number; params: ListsRequestParams }): Promise<{
```

Comment:
this doesn't need to be async from what I can see

Reply:
I get the ESLint error when I remove async: `ESLint: Functions that return promises must be async.(@typescript-eslint/promise-function-async)`

Reply:
still didn't like it so I've reverted back to an interface.